src/app/consumption-estimation/consumption-estimation.component.ts

```diff
@@ -0,0 +1,206 @@
+
+  serviceChoiceForm!:FormGroup;
+  familyAdressCompoundForm!: FormGroup;
+  consumptionDetailsForm!: FormGroup;
+  meterTypeForm!:FormGroup;
+  service_step = false;
+  familyAdress_step = false;
+  meterType_step=false;
+  consumption_step = false;
+  
+  step = 1;
+  selectedMeterNr=1;
+  estimation!:number;
+  equipments: Array<any> = [
+    { name: 'Oven/Stove', value: '1' },
+    { name: 'Dishwasher', value: '2' },
+    { name: 'Washing Machine', value: '3' },
+    { name: 'Drying Machine', value: '4' },
+    { name: 'Hair Dryer', value: '5' }
+  ];
+
+  constructor(private formBuilder: FormBuilder) {}
+
+  ngOnInit() 
+  {
+    this.serviceChoiceForm = this.formBuilder.group({
+      choice:['', Validators.required],
+      
+    });
+    this.familyAdressCompoundForm = this.formBuilder.group({
+        city: ['', Validators.required],
+        postcode: ['',Validators.required],
+        members: ['',Validators.required],
+        houseType: ['',Validators.required]
+    });
+    this.meterTypeForm = this.formBuilder.group({
+      metersNumber: ['', Validators.required],
+      meterTypes: [[],Validators.required],
+      metersValue:[[],Validators.required]
+    });
+    this.consumptionDetailsForm = this.formBuilder.group({
+        yearlyConsumption: ['',Validators.required],
+        selectedEquipments : new FormArray([])
+    });
+  }
+  get service() { return this.serviceChoiceForm.controls; }
+  get familyAddress() { return this.familyAdressCompoundForm.controls; }
+  get meterType() {return this.meterTypeForm.controls;}
+  get consumption() { return this.consumptionDetailsForm.controls; }
+
+  onCheckboxChange(event: any) {
+    const selectedEquipments = (this.consumption['selectedEquipments'] as FormArray);
+    if (event.target.checked) {
+      selectedEquipments.push(new FormControl(event.target.value));
+    } else {
+      const index = selectedEquipments.controls.findIndex(x => x.value === event.target.value);
+      selectedEquipments.removeAt(index);
+    }
+  }
+
+
+next()
+{
+  if(this.step==1)
+  {
+    this.service_step=true;
+    if(this.serviceChoiceForm.invalid){return}
+    this.step++;
+    console.log(this.step);
+  }
+  if(this.step==2)
+  {
+    console.log(this.step);
+    this.familyAdress_step=true;
+    if(this.familyAdressCompoundForm.invalid)
+    {
+      console.error();
+      return
+    }
+    this.step++;
+    console.log(this.step);
+  }
+  if(this.step==3)
+  {
+    console.log(this.step);
+    this.meterType_step=true;
+    if(this.meterTypeForm.invalid)
+    {
+      console.error();
+      return
+    }
+    this.step++;
+    console.log(this.step);
+  }
+}
+previous()
+{
+  this.step--;
+  console.log(this.step);
+  if(this.step==1)
+  {
+    this.service_step=false;
+    console.log(this.step);
+  }
+  if(this.step==2)
+  {
+    this.familyAdress_step=false;
+    console.log(this.step);
+  }
+  if(this.step==3)
+  {
+    this.meterType_step=false;
+    console.log(this.step);
+  }
+  
+}
```

Comment:
Can this be simplified?
[Second example here for showing errors seems more fit perhaps?](https://material.angular.io/components/stepper/examples)